Frameworks/CoreGraphics/CGContextCairo.mm

```diff
@@ -1629,6 +1630,229 @@
     UNLOCK_CAIRO();
 }
 
+void CGContextCairo::_CGContextDrawStrokedPath(cairo_path_t* flatPath, int i, int prev, int total, CGFloat lineWidth) {
+    cairo_path_data_t* data = &flatPath->data[i];
+    int nextIndex = i + flatPath->data[i].header.length;
+    bool moreData = nextIndex < total;
+    // Try to get next and previous types.
+    cairo_path_data_type_t nextType = CAIRO_PATH_CLOSE_PATH;
+    cairo_path_data_type_t prevType = CAIRO_PATH_CLOSE_PATH;
+    if (moreData) {
+        nextType = flatPath->data[nextIndex].header.type;
+    }
+    if (i > 0) {
+        prevType = flatPath->data[prev].header.type;
+    }
+    switch (data->header.type) {
+        case CAIRO_PATH_MOVE_TO: {
+            // Don't do a move here, it'll move to the center of our stroke.
+            // Save the current position for later. This will help us close the current sub path
+            CGPoint lastPos = curPathPosition;
+            // Set the current position
+            curPathPosition.x = data[1].point.x;
+            curPathPosition.y = data[1].point.y;
+
+            // If there's more data process it
+            if (moreData) {
+                _CGContextDrawStrokedPath(flatPath, nextIndex, i, total, lineWidth);
+                // return the current point to the saved point.
+                curPathPosition.x = lastPos.x;
+                curPathPosition.y = lastPos.y;
+            }
+            break;
+        }
+        case CAIRO_PATH_CURVE_TO:
+            // There shouldn't be curves in this path. Move on.
```

Comment:
Yes we could put a warning trace here. We know in this situation that there are no curves, but a warning could help someone in the future.